lib/logger.rb

```diff
@@ -577,15 +597,11 @@ class LogDeviceMutex
       include MonitorMixin
     end
 
-    def initialize(log = nil, opt = {})
+    def initialize(log, opt = {})
```

Comment:
Making an optional argument required seems a backward incompatibility.
Is this needed?